dev/com.ibm.ws.org.jboss.weld/src/org/jboss/weld/resources/FileLogger.java

```diff
@@ -0,0 +1,707 @@
+/*******************************************************************************
+ * Copyright (c) 2009, 2023 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.jboss.weld.resources;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.WeakHashMap;
+
+// Copied from:
+// dev/com.ibm.ws.ras.instrument/src/com/ibm/ws/ras/instrument/internal/main/FileLogger.java
+
+public class FileLogger {
+    public static final String CLASS_NAME = FileLogger.class.getSimpleName();
+
+    // Time utility ...
+    //
+    // Initialization performed here must occur before the creation of the
+    // file logger singleton.  That initialization calls back to 'getTime'.
+    // If the initialization happens after the creation of the singleton,
+    // the call to create the singleton calls back to retrieve the formatted
+    // time before 'current' is assigned, leading to a NPE.
+
+    public static long getTime() {
+        return System.currentTimeMillis();
+    }
+
+    private static final SimpleDateFormat formatter =
+        new SimpleDateFormat("MM/dd/yy HH:mm:ss:SSS z");  // 03/22/16 12:01:13:654 ESD
+
+    private static final Date current = new Date( getTime() );
+    private static String currentFormatted = formatter.format(current);
+
+    public static String getFormattedTime() {
+        long currentMs = getTime();
+        if ((currentMs - current.getTime()) > 10)  {
+            current.setTime(currentMs);
+            currentFormatted = formatter.format(current);
+        }
```

Comment:
In a multi threaded environment this isn't going to be thread safe either.

Reply:
Added sync.